src/SymArrayDmap.chpl

```diff
@@ -92,27 +92,39 @@ module SymArrayDmap {
     }
 
     proc makeDistArray(D: domain(?), type etype) throws
-      where D.rank == 1
+      where MyDmap != Dmap.defaultRectangular && D.isDefaultRectangular()
+    {
+      return makeDistArray(makeDistDom(D), etype);
+    }
+
+    proc makeDistArray(D: domain(?), type etype) throws
+      where D.rank == 1 && (MyDmap == Dmap.defaultRectangular || !D.isDefaultRectangular())
     {
       var res = D.tryCreateArray(etype);
       return res;
     }
 
     proc makeDistArray(D: domain(?), type etype) throws
-      where D.rank > 1
+      where D.rank > 1 && (MyDmap == Dmap.defaultRectangular || !D.isDefaultRectangular())
```

Comment:
This is the only case that gave me a pause. If I am not mistaken, this is for when:

- the server is configured to use local arrays
- and the domain on which we are supposed to create an array is distributed

And on this branch, it looks like that would result in this function returning a distributed array.

If that's the case my reflex is to return a local array here, as I would think that the server's configuration is what matters in terms of the behavior of `makeDistArray`. What do you think?

Reply:
I see your point, I had the same thought when implementing this, but I decided to match the behavior for other overloads of `makeDistArray` which have the same behavior:

https://github.com/Bears-R-Us/arkouda/blob/a723f6f78fc2057cb3dba811e6027b5439271338/src/SymArrayDmap.chpl#L80-L86

If we change this here, we should also change the behavior in other overloads.

At the same time, I don't know if it's really a problem. The problem would be if someone creates the "first" distributed domain/array on the server without regard for the configuration being multi-locale or not. 
This is just a secondary way to make arrays and domains, using other existing domains. So it's not a problem by itself, and in any case, this overload should be kept to cover all the cases of the `where` clauses.

Reply:
https://github.com/Bears-R-Us/arkouda/issues/4243